### ISSUE TITLE
Published 1.5.1 broke public API

### ISSUE THREAD

**Issue by kvark** [gfx-rs/rspirv#196]
https://docs.rs/spirv_headers/1.5.0/spirv_headers/enum.ExecutionMode.html#method.required_capabilities is gone.
See also - https://github.com/gfx-rs/naga/issues/836
cc @Jasper-Bekkers

**Comment by Jasper-Bekkers** [comment]
spirv_headers don't follow semantic versioning so breakage like this is sort-of unavoidable without some changes to the policy of this project :-(

We've run into this before with #117

**Comment by Jasper-Bekkers** [comment]
For reference - required_capabilities and required_extensions got moved into `Operand` instead because there can be multiple capabilties or extensions required.

**Comment by kvark** [comment]
Wait, #117 was actually fixed, and sanity was restored. Are you saying that the current breakage is intended?

**Comment by Jasper-Bekkers** [comment]
In #117 a workaround got introduced to not break semver, however in this case I'm not sure we can (see previous comment).

**Comment by Jasper-Bekkers** [comment]
It's specifically [this](https://github.com/gfx-rs/rspirv/issues/117#issuecomment-548401092) comment that explains why we're in this spot though I'm not a big fan (for obvious reasons).

**Comment by antiagainst** [comment]
That comment was made quite some time ago when we saw the first breakage. Things do change and I'm certainly not trying to hold back improvements. :) We had similar discussions in another thread later (see [here](https://github.com/google/swiftshader/blob/72b1f01722ce8567212f931a189e439c59b8d731/src/Vulkan/libVulkan.cpp#L159)) where I suggested to retire the `spirv_headers` create and use the `spirv` crate instead to have a clean slate for using semver. This might be the time to really push towards that.
